hoot-core/src/main/cpp/hoot/core/scoring/MatchComparator.cpp

```diff
@@ -500,46 +563,64 @@ int MatchComparator::getTotalCount() const
 
 bool MatchComparator::_isNeedsReview(QString uuid1, QString uuid2, const ConstOsmMapPtr& conflated)
 {
-  ElementId eid1 = _actualUuidToEid[uuid1];
-  ElementId eid2 = _actualUuidToEid[uuid2];
+  QList<ElementId> eid1s = _actualUuidToEid.values(uuid1);
+  QList<ElementId> eid2s = _actualUuidToEid.values(uuid2);
 
   bool result = false;
-
-  if (eid1.isNull() || eid2.isNull())
+  for (int eid1Index = 0; eid1Index < eid1s.size(); eid1Index++)
   {
-    //So far this message is ok, change from LOG_INFO to LOG_DEBUG.
-    //More information please see issue 167. https://github.com/ngageoint/hootenanny/issues/167
-    LOG_DEBUG("No actual element exists.");
-    return false;
-  }
+    ElementId eid1 = eid1s.at(eid1Index);
+    for (int eid2Index = 0; eid2Index < eid2s.size(); eid2Index++)
+    {
+      ElementId eid2 = eid2s.at(eid2Index);
+      if (eid1.isNull() || eid2.isNull())
+      {
+        //So far this message is ok, change from LOG_INFO to LOG_DEBUG.
+        //More information please see issue 167. https://github.com/ngageoint/hootenanny/issues/167
+        LOG_DEBUG("No actual element exists.");
+        return false;
+      }
 
-  if (ReviewMarker().isNeedsReview(conflated, conflated->getElement(eid1),
-    conflated->getElement(eid2)))
-  {
-    result = true;
+      if (ReviewMarker().isNeedsReview(conflated, conflated->getElement(eid1),
+        conflated->getElement(eid2)))
+      {
+        result = true;
+      }
+    }
   }
-
   return result;
 }
 
 void MatchComparator::_tagError(const OsmMapPtr &map, const QString &uuid, const QString& value)
 {
-  // if the uuid contains the first uuid
-  TagContainsFilter tcf(Filter::KeepMatches, "uuid", uuid);
-  // set mismatch to 1.
+  // if the uuid contains the first uuid, set mismatch
   SetTagVisitor stv("hoot:mismatch", value);
-  FilteredVisitor fv(tcf, stv);
-  map->visitRw(fv);
+  MatchComparator::UuidToEid::iterator it = _actualUuidToEid.begin();
```

Comment:
Is _actualUuidToEid a MultiMap of string to uuids here, or is it something else? Could you use _actualUuidToEid.values(uuid) to just get a list of the values you are interested in, instead of iterating over all of _actualUuidToEid ?